### PR TITLE
Add metrics schema

### PR DESCRIPTION
**Pull request by Nicoretti**

### CODE REVIEW

File: metrics-0.1.0.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      "anyOf": [
+        {
+          "maximum": 100.0,
+          "minimum": 0.0,
+          "type": "number"
+        },
+        {
+          "type": "string"
+        }
+      ],
```
Comment (kaklakariada) [inline metrics-0.1.0.json:40]: Coverage should be only a number, not a string.
If possible add a description/title with a bit more description, e.g. "Line coverage of tests in percent"

File: metrics-0.1.0.json
Comment (kaklakariada) [inline metrics-0.1.0.json:1]: General: the name "metrics" is too generic without prior knowledge. Please rename to something more specific, e.g. project-quality-metrics

File: metrics-0.1.0.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    "reliability": {
+      "$ref": "#/$defs/Rating"
+    },
+    "security": {
+      "$ref": "#/$defs/Rating"
+    },
+    "technical_debt": {
+      "$ref": "#/$defs/Rating"
+    }
```
Comment (kaklakariada) [inline metrics-0.1.0.json:55]: Add a description to each field. References to the "Rating" enum are not rendered in Bootprint: https://exasol.github.io/schemas/metrics-0.1.0/index.html

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: metrics-0.1.0.json (hunk 1 above)
Comment (Nicoretti) [reply metrics-0.1.0.json]: Fixed

File: metrics-0.1.0.json (hunk 2 above)
Comment (Nicoretti) [reply metrics-0.1.0.json]: As discussed this seems to be a limitation of the tool we are using right now (for details see #25).

File: metrics-0.1.0.json
Comment (Nicoretti) [reply metrics-0.1.0.json:1]: I personally would like to keep metrics at least for now, or is this a deal breaker for you?

File: metrics-0.1.0.json [hunk 3]
```diff
@@ -19,39 +19,54 @@
   "description": "This schema defines the structure and values for reporting Q/A metrics for projects.",
   "properties": {
     "commit": {
+      "description": "Commit-Hash pointing to the state of the codebase used for generating the metrics.",
```
Comment (kaklakariada) [inline metrics-0.1.0.json:22]: Would it make sense to add a field for the branch where the metrics where generated? E.g. `main` or `feature/xyz`? Or will it always be the main branch?

File: metrics-0.1.0.json
Comment (kaklakariada) [reply metrics-0.1.0.json:1]: I really would prefer something like "code quality metrics", "code metrics" or "source code metrics" to make it clear what this is about.

File: metrics-0.1.0.json [hunk 4]
```diff
@@ -19,39 +19,54 @@
   "description": "This schema defines the structure and values for reporting Q/A metrics for projects.",
```
Comment (kaklakariada) [inline metrics-0.1.0.json:19]: Please also add a description for the `Rating`, mentioning that `A` is good and `F` is bad, to avoid misunderstandings.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

File: metrics-0.1.0.json (hunk 3 above)
Comment (Nicoretti) [reply metrics-0.1.0.json]: A branch isn't a unique point in time a commit hash is. Also at specific points in time multiple branches can point to the same sha1 (commit hash).

**Pull request merged**